taichi/program/program.cpp

```diff
@@ -509,6 +510,10 @@ void Program::materialize_snode_tree(SNodeTree *tree) {
   }
 }
 
+void Program::destroy_snode_tree(const int snode_tree_id) {
+  memory_pool->destroy_snode_tree(snode_tree_id);
```

Comment:
This is not the correct level of abstraction. Memory pool should only care about memory allocation. I'd suggest to have a  separate class to manage SNodeTree's memory.

Reply:
Thanks for your suggestions. I'll rewrite this part of code.